bolt-simple-taproot.md

```diff
@@ -0,0 +1,943 @@
+    aggregated public key.
+
+  3. Next, each party generates a _partial_ signature and exchanges it with
+    every other party.
+
+  4. Finally, once all partial signatures are known, they can be combined into a
+    valid schnorr signture, which can be veirfied using BIP 340.
+
+Steps 1&2 may be performed out of order, or concurrently. In our case only two
+parties exist, so as soon as one party knows both partial signatures, they can
+be combined into a final signature.
+
+#### Key Aggregation
+
+We refer to the of `KeyAgg` algorithm [bip-musig2](https://github.com/jonasnick/bips/blob/musig2/bip-musig2.mediawiki)
+for key aggregation. In order to avoid sending extra ordering information, we
+always assume that both keys are sorted first with the `KeySort` algorithm
+before aggregating (`KeyAgg(KeySort(p1, p2))`)>
+
+#### Nonce Generation
+
+A `musig2` public nonce is a 66-byte value which is the concentration of two EC
+points encoded using the compressed format:
+```
+nonce_1 || nonce_2
+```
+
+Each public nonce has a corresponding _secret_ nonce which is only known to the
+signer.
+
+For nonce generation, we refer to the `NonceGen` algorithm defined in the
+`musig2` BIP. We only specify the set of required arguments, though a signer
+can also opt to specify the additional argument sin order to strengthen their
+nonce.
+
+#### Nonce Handling
+
+Throughput this document we always assume that _fresh_ nonces are generated for
+each signing attempt, and nonces are **never** written to disk. As we'll see
+later, this has implications with respect to the retransmission protocol.
+
+Note that nonces can safely be pre-generated, and even batched/pipelined ahead
+of time to reduce the number of round trips in the protocol.
+
+Finally, due to the asymmetric state of the current revocation channels, we
+actually require two sets of public nonces to be exchanged by each party: one
+for the local commitment, and one for the remote commitment.
+
+#### Signing
+
+Once public nonces have been exchanged, the `NonceAgg` algorithm is used to
+combine the public nonces into the combined nonce which will be a part of the
+final combined signature.
+
+The `Sign` is used to generate a valid partial signature, with the
+`PartialSigVerify` algorithm used to verify each signature.
+
+Once all partial signatures are known, the `PartialSigAgg` algorithm is used to
+combine the partial signature into a final valid BIP 340 schnorr signature.
+
+### Nothing Up My Sleeves Points
+
+Whenever we want to ensure that a given P2TR can _only_ be spent via a script
+path, we utilize a Nothing Up My Sleeve (NUMs) point. A NUMs point is an EC
+point that no one knows the private key to. If no one knows the private key,
+then it can't be used for key path signing, forcing the script path to always
+be taken.
+
+We refer to the `simple_taproot_nums` as the following value:
+```
+02dca094751109d0bd055d03565874e8276dd53e926b44e3bd1bb6bf4bc130a279
```

Comment:
https://github.com/lightning/bolts/pull/995#issuecomment-1144469224

dont use a NUMS point but the revocation key for less fees and more privacy

Reply:
https://github.com/lightning/bolts/pull/995#issuecomment-1167871874

or dont use a 1 csv locked to_remote output

Reply:
you want a 1 csv to prevent the remote party seeing the commit tx in the mempool and pinning it

Reply:
in my comment i explained how this would work, by removing the remote anchor and removing the 1 csv from the to_remote output the remote party still has 1 output to fee bump and pin if he wants just like with the anchor. pinning makes rbf hard, it hasnt anything to do with cpfp

Reply:
Removing the anchor will work - I hadn't read your comment, so my comment can be ignored. But pinning does have to do with CPFP since it deals with the mempool and its package limits. However, I still don't think the change is worth adding more logic either at the spec or code level

Reply:
Yeh it's gone now in the latest diff. I agree with @Crypt-iQ not sure it's worth the trouble to re-work the anchors significantly. The latest diff re-uses the multi-sig key.